tests/utils/date-service.test.ts

```diff
@@ -0,0 +1,44 @@
+import { getTimeTrackedInSeconds } from '@utils/date-service';
+
+describe('date-service tests', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('getIssueWorkedSeconds method', () => {
+    it('should return a positive value if no end time provided', async (): Promise<void> => {
+      // given
+      const startTime: Date = new Date('2024-03-12T09:00:00Z');
+      const zeroSeconds: number = 0;
+      // when
+      const receivedSeconds: number = getTimeTrackedInSeconds(startTime, null);
+      // then
+      expect.assertions(1);
+      expect(receivedSeconds).toBeGreaterThan(zeroSeconds);
+    });
+
+    it('should return a positive value under normal conditions', async (): Promise<void> => {
+      // given
+      const startTime: Date = new Date('2024-03-12T09:00:00Z');
+      const endTime: Date = new Date('2024-03-12T10:00:00Z');
+      const expectedSeconds: number = 3600;
+      // when
+      const receivedSeconds: number = getTimeTrackedInSeconds(startTime, endTime);
+      // then
+      expect.assertions(1);
+      expect(receivedSeconds).toEqual(expectedSeconds);
+    });
+
+    it('should return a negative value if tracking time record is not valid', async (): Promise<void> => {
```

Comment:
Negative time doesn't exist. If this is not possible, this is an exceptional possibility so you should throw an exception to handle the error properly. Again, this shouldn't happen, but if it happened, the error should be managed here and not in the service, as this function is the one in charge of making the calculations.

Reply:
fixed